### DISCUSSION TITLE
Welcome to EcoLogits! (looking for feedbacks)

### DISCUSSION THREAD

**Discussion by samuelrince** [mlco2/ecologits#45] [Announcements]
## :wave: Welcome to EcoLogits

We're thrilled to announce the initial release of our python package 🌱 **EcoLogits**. This package has been developed with the aim to simplify the tracking of the energy consumption and the environmental footprint of generative AI models at inference. 

As the first set of users, your experience, feedback, and suggestions can play a crucial role in the further development and refinement of EcoLogits. We're eager to hear about:

* Your general feedback and first impressions of the package.
* Any issues or bugs you've encountered during installation or use.
* Any additional features or enhancements you'd like to see in future releases.
* The effectiveness of our documentation and examples.

Also, feel free to:

* Ask any questions or queries related to the package.
* Share your ideas and use-cases where you've implemented EcoLogits.

Let's make EcoLogits better together! And a warm thank you for all the feedbacks! 🙏

**Comment by Marie-PerrotDockes** [comment]
Hi Samuel,

Thank you for the great app! 

I have a question on the expert mode when you compare the impact of an LLM in different countries:  the values are obtained as if all the energy used to run an LLM in a given country was produced in this country? Is that the case? (If I ask a question to a LLM in France is it running on a datacenter in France ?)

Best and thank you,

**Comment by samuelrince** [comment]
Hello @Marie-PerrotDockes,

Thank you for the feedback! You are right, when you change the location it is used to change to the according electricity mix of the selected country. This concerns only the energy consumed by the hardware and cooling inside the data center. So, this basically if you choose France, then it means the data center should be in France.

**Comment by aminaslo** [comment]
Hello! I would like to include the EcoLogits calculator in an AI usage guidance policy for my organization. I would love if it also provided calculations for image generation; is this something that could be added in the list of sample prompts?

**Reply by samuelrince** [reply]
Hi @aminaslo, unfortunately we lack comprehensive energy benchmarks of open models for image generation to add it to EcoLogits. For now there is no plan to implement it in the coming months, as this is a lot of work. 

Even though newer models are currently not available, you can check the [ML.ENERGY Leaderboard](https://ml.energy/leaderboard/?__theme=light) to see real-world energy consumption metrics per prompt.

**Reply by aminaslo** [reply]
I see. That is unfortunate, but thank you for the explanation and the tool!

**Comment by oddskool** [comment]
Hello,

Thanks for the great work. I have a question regarding training emissions. 

AFAICT ecologits does not estimate emissions during training of LLMs, right ?

If so I believe it would be best to mark it clearly in the Calculator so that people don't underestimate this impact (which I guess is orders of magnitude larger than inference ?).

**Reply by adrienbanse** [reply]
Hey @oddskool, thanks for the kind words. Concerning training emissions, see our previous answer: 

> We do not plan (in the short term, at least) to support training impacts in EcoLogits. The methodology is designed to assess impacts at inference and is intended to address big providers like OpenAI. To include a part of the training impacts into each request, we would need "trade secret" information from these providers about the training impacts but also the usage of their services.
> 
> For instance, Meta computed the energy consumption and carbon emissions (1900 tCO2eq) (only derived from direct energy consumption) for Llama 3 70B [source](https://huggingface.co/meta-llama/Meta-Llama-3-70B-Instruct#hardware-and-software). If we were to include it in our methodology for inference, we would somehow need to divide this impact among all the requests that will ever be made to this model. But even estimating this number is very difficult.
> 
> Plus, we quickly realize that for models deployed at that scale, dividing ~2000 tCO2eq or even 10,000 tCO2eq by billions of requests made by millions of users is negligible compared to the impact of simply running the model.
> 
> What we can consider adding in the future is a customizable way of accounting for the training impacts if you know your own usage as a company, for instance, but it's not really on the roadmap for now. ;)

In short: we do not account for training impact and think that they are orders of magnitude **lower** than inference. 

The fact that we account only for inference is written in the methodology toggle of the calculator, but you're right that we should maybe put it more clearly! Thanks for the feedback.
